### ISSUE TITLE
UI: "Unexpected end of JSON input" when selecting FIDO2/WebAuthn dropdown

### ISSUE THREAD

**Issue by skrobul** [mailcow/mailcow-dockerized#4438]
**Prior to placing the issue, please check following:** *(fill out each checkbox with an `X` once done)*
- [x] I understand that not following or deleting the below instructions will result in immediate closure and/or deletion of my issue.
- [x] I have understood that this bug report is dedicated for bugs, and not for support-related inquiries.
- [x] I have understood that answers are voluntary and community-driven, and not commercial support.
- [x] I have verified that my issue has not been already answered in the past. I also checked previous [issues](https://github.com/mailcow/mailcow-dockerized/issues).

## Summary

On the login page, selecting selecting FIDO2/WebAuthn dropdown (next to Login) button results in `"Validation failed: Unexpected end of JSON input"` message. Sometimes it's also `"JSON.parse: unexpected end of data at line 1 column 1 of the JSON data"`. The browser does not ask for key input either.
Same behaviour with Firefox and Chromium.

I am using version tagged with `2022-01` (5a1ef72b8225837d63aa668beb421503c5a30b44).

## Logs

```
Jan 26 18:22:26 my.server.com 805f97fa55d1[2748]: 172.22.1.10 -  26/Jan/2022:18:22:26 +0000 "HEAD /forwardinghosts.php" 200
Jan 26 18:22:26 my.server.com fd773a3659ed[2748]: 172.22.1.12 - - [26/Jan/2022:18:22:26 +0000] "HEAD /forwardinghosts.php HTTP/1.1" 200 0 "-" "rspamd-3.1"
Jan 26 18:22:26 my.server.com 805f97fa55d1[2748]: fd4d:6169:6c63:6f77::f -  26/Jan/2022:18:22:26 +0000 "GET /forwardinghosts.php" 200
Jan 26 18:22:26 my.server.com fd773a3659ed[2748]: 172.22.1.12 - - [26/Jan/2022:18:22:26 +0000] "GET /forwardinghosts.php HTTP/1.1" 200 27 "-" "rspamd-3.1"
Jan 26 18:22:28 my.server.com 805f97fa55d1[2748]: 172.22.1.10 -  26/Jan/2022:18:22:27 +0000 "GET /index.php" 200
Jan 26 18:22:28 my.server.com fd773a3659ed[2748]: 80.3.14.24 - - [26/Jan/2022:18:22:28 +0000] "GET / HTTP/1.0" 200 23152 "-" "Mozilla/5.0 (X11; Fedora; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36"
Jan 26 18:22:28 my.server.com 805f97fa55d1[2748]: fd4d:6169:6c63:6f77::f -  26/Jan/2022:18:22:28 +0000 "GET /json_api.php" 200
Jan 26 18:22:28 my.server.com fd773a3659ed[2748]: 80.3.14.24 - - [26/Jan/2022:18:22:28 +0000] "GET /api/v1/get/passwordpolicy/html HTTP/1.0" 200 0 "https://mail.server.com/" "Mozilla/5.0 (X11; Fedora; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36"
```

What is not immediately obvious from the logs above is that browser sends GET request to `/api/v1/get/fido2-get-args` that unfortunately comes back with 200 OK but has empty body (headers are still present):

request:
```
GET /api/v1/get/fido2-get-args HTTP/1.1
Host: mail.server.com
Connection: keep-alive
Cache-Control: max-age=0
sec-ch-ua: ";Not A Brand";v="99", "Chromium";v="94"
sec-ch-ua-mobile: ?0
User-Agent: Mozilla/5.0 (X11; Fedora; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36
sec-ch-ua-platform: "Linux"
Accept: */*
Sec-Fetch-Site: same-origin
Sec-Fetch-Mode: cors
Sec-Fetch-Dest: empty
Referer: https://mail.server.com/
Accept-Encoding: gzip, deflate, br
Accept-Language: en-GB,en-US;q=0.9,en;q=0.8
Cookie: PHPSESSID=50ca2ab499b4be79a8c3f74d0bd5110b
```
and response:

```
HTTP/1.1 200 OK
Server: nginx
Date: Wed, 26 Jan 2022 18:25:33 GMT
Content-Type: application/json
Transfer-Encoding: chunked
Connection: keep-alive
Vary: Accept-Encoding
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate
Pragma: no-cache
Strict-Transport-Security: max-age=15768000;
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
X-Robots-Tag: none
X-Download-Options: noopen
X-Frame-Options: SAMEORIGIN
X-Permitted-Cross-Domain-Policies: none
Referrer-Policy: strict-origin
```

Obviously, an empty response cannot be parsed as valid JSON.

## Reproduction

Step 1: Navigate to a mailcow UI. You can use official demo at https://demo.mailcow.email/
Step 2: Click on "Login | v" dropdown
Step 3: Select FIDO2/WebAuthn

## System information

| Question | Answer |
| --- | --- |
| My operating system | Fedora 34 |
| Is Apparmor, SELinux or similar active? | yes |
| Virtualization technlogy (KVM, VMware, Xen, etc - **LXC and OpenVZ are not supported** | n/a |
| Server/VM specifications (Memory, CPU Cores) | n/a |
| Docker Version (`docker version`) | n/a |
| Docker-Compose Version (`docker-compose version`) | n/a |
| Reverse proxy (custom solution) | nginx on my instance, not sure about official demo instance |

- Output of `git diff origin/master`, any other changes to the code? **None**.

**Comment by DerLinkman** [comment]
@FreddleSpl0it

**Comment by FreddleSpl0it** [comment]
i cannot reproduce that, but this issue pointed me to a mistake i made [#4440](https://github.com/mailcow/mailcow-dockerized/pull/4440). Maybe it will fix it.

**Comment by skrobul** [comment]
> i cannot reproduce that:

Interesting - this may be specific to a browser(s)/OS the system. This happens on:
- Firefox 94 running on Fedora Linux
- Firefox 96.2 on Android
- Chrome 97.0.4692.98 on Android
- Chrome 97 on Windows

Presumably it would be happening on every other browser though since the empty response comes from the server and is browser-independent. What might be unique about my situation though is that I do NOT have the FIDO2 key inserted on any of the machines I have tested this on (I am new to mailcow and wrongly assumed that this is button is used to login with two factor).

**Comment by FreddleSpl0it** [comment]
> What might be unique about my situation though is that I do NOT have the FIDO2 key inserted on any of the machines I have tested this on (I am new to mailcow and wrongly assumed that this is button is used to login with two factor).

Ok thanks, now i can reproduce that error. I'll have a look.

**Comment by FreddleSpl0it** [comment]
issue should be solved [#4440](https://github.com/mailcow/mailcow-dockerized/pull/4440)

**Comment by milkmaker** [comment]
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.
